lib/stages/visSharedMemWriter.cpp

```diff
@@ -300,7 +298,10 @@ void visSharedMemWriter::main_thread() {
     frame_size = _member_alignment(data_size + metadata_size + 4, alignment * 1024);
 
     // memory_size should be ntime * nfreq * file_frame_size (data + metadata)
-    buf_addr = assign_memory(fname_buf, ntime * nfreq * frame_size);
+    buf_addr = assign_memory(fname_buf, (ntime * access_record_size) + (ntime * nfreq * frame_size));
```

Comment:
- this is missing the valid field, I think
- We need an access record per time per frequency